ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,611 @@
+			mbrMap := networkSecurityGroupMemberToMap(mbr)
+			members = append(members, mbrMap)
+		}
+		d.Set(Attr_Members, members)
+	}
+	d.Set(Arg_Name, networkSecurityGroup.Name)
+	d.Set(Attr_NetworkSecurityGroupID, networkSecurityGroup.ID)
+	if networkSecurityGroup.Rules != nil {
+		rules := []map[string]interface{}{}
+		for _, rule := range networkSecurityGroup.Rules {
+			ruleMap := networkSecurityGroupRuleToMap(rule)
+			rules = append(rules, ruleMap)
+		}
+		d.Set(Attr_Rules, rules)
+	}
+
+	return nil
+}
+
+func resourceIBMPINetworkSecurityGroupRuleDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId("")
+	return nil
+}
+
+func isWaitForIBMPINetworkSecurityGroupRuleAdd(ctx context.Context, client *instance.IBMPINetworkSecurityGroupClient, id, memberID string, timeout time.Duration) (interface{}, error) {
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Pending},
+		Target:     []string{State_Available},
+		Refresh:    isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client, id, memberID),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Minute,
+	}
+
+	return stateConf.WaitForStateContext(ctx)
+}
+
+func isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client *instance.IBMPINetworkSecurityGroupClient, id, memberID string) retry.StateRefreshFunc {
+
+	return func() (interface{}, string, error) {
+		networkSecurityGroup, err := client.Get(id)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if networkSecurityGroup.Members != nil {
+			for _, mbr := range networkSecurityGroup.Members {
+				if *mbr.ID == memberID {
+					return networkSecurityGroup, State_Available, nil
+				}
+
+			}
+		}
+		return networkSecurityGroup, State_Pending, nil
+	}
+}
+
+func isWaitForIBMPINetworkSecurityGroupRuleRemove(ctx context.Context, client *instance.IBMPINetworkSecurityGroupClient, id, memberID string, timeout time.Duration) (interface{}, error) {
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Pending},
+		Target:     []string{State_Removed},
+		Refresh:    isIBMPINetworkSecurityGroupRuleRemoveRefreshFunc(client, id, memberID),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Minute,
+	}
+
+	return stateConf.WaitForStateContext(ctx)
+}
+
+func isIBMPINetworkSecurityGroupRuleRemoveRefreshFunc(client *instance.IBMPINetworkSecurityGroupClient, id, memberID string) retry.StateRefreshFunc {
+
+	return func() (interface{}, string, error) {
+		networkSecurityGroup, err := client.Get(id)
+		if err != nil {
+			return nil, "", err
+		}
+
+		if networkSecurityGroup.Members != nil {
+			isMember := false
+			for _, mbr := range networkSecurityGroup.Members {
+				if *mbr.ID == memberID {
+					isMember = true
+					return networkSecurityGroup, State_Pending, nil
+				}
+			}
+			if !isMember {
+				return networkSecurityGroup, State_Removed, nil
+			}
+		}
+		return networkSecurityGroup, State_Pending, nil
+	}
+}
```

Comment:
Why are you checking for member if this is after removing a rule?

Reply:
I copy pasted and forgot to replace them.